neon_data_models/models/api/llm.py

```diff
@@ -0,0 +1,194 @@
+    enabled: bool = Field(
+        True, description="Flag used to mark a defined persona as "
+                          "available for use.")
+    user_id: Optional[str] = Field(
+        None, description="`user_id` of the user who created this persona.")
+
+    @model_validator(mode='after')
+    def validate_request(self):
+        assert any((self.description, self.system_prompt))
+        if self.system_prompt is None:
+            self.system_prompt = self.description
+        return self
+
+    @computed_field
+    @property
+    def id(self) -> str:
+        persona_id = self.name
+        if self.user_id:
+            persona_id += f"_{self.user_id}"
+        return persona_id
+
+
+class LLMRequest(BaseModel):
+    query: str = Field(description="Incoming user prompt")
+    # TODO: History may support more options in the future
+    history: List[Tuple[Literal["user", "llm"], str]] = Field(
+        description="Formatted chat history (excluding system prompt). Note "
+                    "that the roles used here will differ from those used in "
+                    "OpenAI-compatible requests.")
+    persona: LLMPersona = Field(
+        description="Requested persona to respond to this message")
+    model: str = Field(description="Model to request")
+    max_tokens: int = Field(
+        default=512, ge=64, le=2048,
+        description="Maximum number of tokens to include in the response")
+    temperature: float = Field(
+        default=0.0, ge=0.0, le=1.0,
+        description="Temperature of response. 0 guarantees reproducibility, "
+                    "higher values increase variability. Must be `0.0` if "
+                    "`beam_search` is True")
+    repetition_penalty: float = Field(
+        default=1.0, ge=1.0, le=2.0,
+        description="Repetition penalty. Higher values limit repeated "
+                    "information in responses")
+    stream: bool = Field(
+        default=None, description="Enable streaming responses. "
+                                  "Mutually exclusive with `beam_search`.")
+    best_of: int = Field(
+        default=1,
+        description="Number of beams to use if `beam_search` is enabled.")
+    beam_search: bool = Field(
+        default=None, description="Enable beam search. "
+                                  "Mutually exclusive with `stream`.")
+    max_history: int = Field(
+        default=2, description="Maximum number of user/assistant "
+                               "message pairs to include in history context.")
+
+    @model_validator(mode='before')
+    @classmethod
+    def validate_inputs(cls, values):
+        # Neon modules previously defined `user` and `llm` keys, but Open AI
+        # specifies `assistant` in place of `llm` and is the de-facto standard
+        for idx, itm in enumerate(values.get('history', [])):
+            if itm[0] == "assistant":
+                values['history'][idx] = ("llm", itm[1])
```

Comment:
Just thinking that potentially history element can contain more than 2 values (can ignore)